lib/rules/declaration-block-no-redundant-longhand-properties/index.js

```diff
@@ -80,21 +86,69 @@ const rule = (primary, secondaryOptions) => {
 				for (const shorthandProperty of shorthandProperties) {
 					const prefixedShorthandProperty = prefix + shorthandProperty;
 					const longhandDeclaration = longhandDeclarations.get(prefixedShorthandProperty) || [];
+					const longhandDeclarationNode =
+						longhandDeclarationNodes.get(prefixedShorthandProperty) || [];
 
 					longhandDeclaration.push(prop);
 					longhandDeclarations.set(prefixedShorthandProperty, longhandDeclaration);
 
-					const shorthandProps = /** @type {Map<string, Set<string>>} */ (
-						longhandSubPropertiesOfShorthandProperties
-					).get(shorthandProperty);
+					longhandDeclarationNode.push(decl);
+					longhandDeclarationNodes.set(prefixedShorthandProperty, longhandDeclarationNode);
+
+					const shorthandProps =
+						/** @type {import('stylelint').OrderedLonghandSubPropertiesOfShorthandProperties} */ (
+							orderedLonghandSubPropertiesOfShorthandProperties
+						).get(shorthandProperty);
 					const prefixedShorthandData = Array.from(shorthandProps || []).map(
 						(item) => prefix + item,
 					);
 
-					if (!arrayEqual(prefixedShorthandData.sort(), longhandDeclaration.sort())) {
+					const copiedPrefixedShorthandData = [...prefixedShorthandData];
+
+					if (!arrayEqual(copiedPrefixedShorthandData.sort(), longhandDeclaration.sort())) {
 						continue;
 					}
 
+					if (context.fix) {
+						const declNodes = longhandDeclarationNodes.get(prefixedShorthandProperty) || [];
+
+						if (declNodes !== undefined && declNodes[0] !== undefined) {
+							const resolveShorthandValues = () => {
```

Comment:
Should I pull out this function? The nesting is pretty nasty; however, I wasn't sure if separating it (and adding several arguments) would make the function less readable; this version is "in-context".

Reply:
opinion

Personally I prefer having a separate fixer function at l110, if it doesn't have too many arguments. In your case there's too many so it would have to be in the same scope.

The reason being that it prepares the refactor needed for https://github.com/stylelint/stylelint/issues/2643#issuecomment-535406358

Reply:
Ah, this is super clean - thank you for this suggestion! Also excited to use optional chaining more frequently :)